src/components/DayPickerRangeController.jsx

```diff
@@ -144,10 +146,13 @@ export default class DayPickerRangeController extends React.Component {
 
       if (!startDate) {
         endDate = day;
-        this.props.onFocusChange(START_DATE);
+        onFocusChange(START_DATE);
       } else if (isInclusivelyAfterDay(day, firstAllowedEndDate)) {
         endDate = day;
-        if (!keepOpenOnDateSelect) this.props.onFocusChange(null);
+        if (!keepOpenOnDateSelect) {
+          onFocusChange(null);
+          this.props.onClose({ startDate, endDate });
```

Comment:
You can always rebase down to an atomic set of commits; please do so frequently :-)